File: JWSPLIN-wspolliniowosc/JWSPLIN.rb
n=gets().to_i
n.times do |i| 
    ax,ay,bx, by, cx,cy=gets().split("\t").map(&:to_i)
    w=ax*by+bx*cy+cx*ay-by*cx-cy*ax-bx*ay
    if w== 0 then 
    	print "TAK\n"
    else
    	print "NIE\n"
    end
end

File: JWSPLIN-wspolliniowosc/JWSPLIN.cpp
#include <iostream>

using namespace std;

int main(){
    int t;
    int w;
    int ax,ay,bx,by,cx,cy;
    cin >>t;
    while(t--){
        cin>>ax>>ay>>bx>>by>>cx>>cy;
        w=ax*by+bx*cy+cx*ay-by*cx-cy*ax-bx*ay;
        if(w==0){
            cout<<"TAK"<<endl;
        } else {
            cout<<"NIE"<<endl;
        }
    }
}